libr/debug/p/native/linux/linux_debug.c

```diff
@@ -408,10 +409,26 @@ static int linux_stop_process(int pid) {
 	return ret == pid;
 }
 
-static int linux_attach_single_pid(RDebug *dbg, int ptid) {
-	linux_set_options (dbg, ptid);
-	return r_debug_ptrace (dbg, PTRACE_ATTACH, ptid, NULL,
-		(r_ptrace_data_t)(size_t)NULL);
+static bool linux_attach_single_pid(RDebug *dbg, int ptid) {
+	siginfo_t sig = {0};
+	// Safely check if the PID has already been attached to avoid printing errors.
+	// Attaching to a process that has already been started with PTRACE_TRACEME.
+	// sets errno to "Operation not permitted" which may be misleading.
+	// GETSIGINFO can be called multiple times and would fail without attachment.
+	if (-1 == r_debug_ptrace(dbg, PTRACE_GETSIGINFO, ptid, NULL,
```

Comment:
missing space before ( and piut the == -1 in the right side of the statement

Reply:
Updated